### PR TITLE
Update Maven wrapper and fix Quarkus CLI test failures in daily builds

### PR DESCRIPTION
**Pull request by michalvavrik**
### Summary

Apache Maven and wrapper has been recently updated in upstream - https://github.com/quarkusio/quarkus/pull/31836, our CI is creating Quarkus CLI from JARs created by this wrapper and when we run `./quarkus-dev-cli build`, it is using our wrapper from Quarkus Test Suite root directory which is old one and incompatible with the one that was used for bulding CLI jar. I opened discussion here https://github.com/quarkusio/quarkus/issues/32081#issuecomment-1484208999 to confirm it is expected behavior (and documented). All files I added are copy & paste from Quarkus main project.

Please select the relevant options.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency update
- [ ] Refactoring
- [ ] Backport
- [ ] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

File: .mvn/jvm.config [hunk 1]
```diff
@@ -0,0 +1,4 @@
+-Dmaven.wagon.httpconnectionManager.ttlSeconds=120
+-Dmaven.wagon.http.retryHandler.requestSentEnabled=true
+-Dmaven.wagon.http.retryHandler.count=10
+-Xmx5g
```
Comment (rsvoboda) [inline .mvn/jvm.config:4]: why is this set?

Comment (michalvavrik) [reply .mvn/jvm.config]: I planned to use same settings as Quarkus does, but it seems like they only needs this to fix transient issues on Azure, so we should be fine. I dropped it.

**Comment by michalvavrik** [comment]
@rsvoboda it looks like it is either getting fixed / reverted in upstream. it seems like a good thing we had this old wrapper as long as it's supported. let's wait little with this, if there is no fix tomorrow, I can disable Quarkus CLI test that is failing.

**Comment by michalvavrik** [comment]
@rsvoboda looks like we have final decision now - 3.6.2 won't be supported now - https://github.com/quarkusio/quarkus/issues/32081#issuecomment-1485314049, so we need to do some upgrade in order to fix CI. This PR would do, let me know if you want something different.

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**